src/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs

```diff
@@ -0,0 +1,510 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.DirectoryServices.Tests;
+using System.Linq;
+using Xunit;
+
+namespace System.DirectoryServices.AccountManagement.Tests
+{
+    public class AccountManagementTests
+    {
+        internal static bool IsLdapConfigurationExist => LdapConfiguration.Configuration != null;
+        internal static bool IsActiveDirectoryServer => IsLdapConfigurationExist && LdapConfiguration.Configuration.IsActiveDirectoryServer;
+
+        [ConditionalFact(nameof(IsActiveDirectoryServer))]
+        public void TestCurrentUse()
+        {
+            using (PrincipalContext context = DomainContext)
+            using (UserPrincipal p = FindUser(LdapConfiguration.Configuration.UserNameWithNoDomain, context))
+            {
+                Assert.NotNull(p);
+                Assert.Equal(LdapConfiguration.Configuration.UserNameWithNoDomain, p.Name);
+
+                using (UserPrincipal cu = UserPrincipal.Current)
+                {
+                    try
```

Comment:
.Current requires a domain joined client so we might want to mark that in the method dependencies.